account_commission_sales_fee/models/commission_settlement.py

```diff
@@ -11,7 +11,10 @@ def _prepare_sales_fee_product(self, commission):
         return {
             "product_id": commission.sales_fee_product_id.id,
             "quantity": len(
-                self.line_ids.filtered(lambda line: line.commission_id == commission)
+                self.line_ids.filtered(
+                    lambda line: line.commission_id == commission
+                    and line.invoice_line_id.move_id.move_type == "out_invoice"
+                )
```

Comment:
I don't think we should rely on move_type as out_invoice could also hold lines with negative subtotals.
```suggestion
                self.line_ids.filtered(
                    lambda line: line.commission_id == commission and line.settled_amount > 0.0
                )
```

Reply:
I updated the code.